_15thnight/api/need.py

```diff
@@ -24,22 +23,27 @@ def get_need(need_id):
 @need_api.route('/need/<int:need_id>/resolve', methods=['POST'])
 @required_access('advocate')
 def resolve_need(need_id):
+    """
+    Resolve a need and close an alert if necessary.
+    Send out a message stating the alert was closed as well.
+    """
     need = Need.get(need_id)
+
+    # Check validity of need_id
     if not need:
         return api_error('Need not found')
     if current_user.id != need.alert.user_id:
         return api_error('Permission denied')
     if need.resolved:
         return api_error('Need already resolved!')
+
+    # valid need_id given, validate the form
     form = ResolveNeedForm(need=need)
     if not form.validate_on_submit():
         return api_error(form.errors)
-    need.resolved = True
-    need.resolved_at = datetime.utcnow()
-    need.resolve_notes = form.notes.data
-    need.resolve_message = form.message.data
-    for provision in need.provisions:
-        provision.selected = provision.id in form.provisions.data
-    send_out_resolution(need)
-    need.save()
-    return '', 200
\ No newline at end of file
+
+    # Check if alert is resolved, notify those involved
+    queue_resolve_alert.apply_async()
```

Comment:
The pattern the project is following a pattern of calling functions from `core`. `core` does asynchronous things. Please leave this as `send_out_resolution`.